libcst/codemod/commands/rename.py

```diff
@@ -0,0 +1,192 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+#
+# pyre-strict
+import argparse
+from typing import Dict, Optional, Sequence, Union
+
+import libcst as cst
+from libcst.codemod import CodemodContext, VisitorBasedCodemodCommand
+from libcst.codemod.visitors import AddImportsVisitor, RemoveImportsVisitor
+from libcst.helpers import get_full_name_for_node
+from libcst.metadata import QualifiedNameProvider
+
+
+class RenameCommand(VisitorBasedCodemodCommand):
+    """
+    Rename all instances of a local or imported object.
+    """
+
+    DESCRIPTION: str = "Rename all instances of a local or imported object."
+
+    METADATA_DEPENDENCIES = (QualifiedNameProvider,)
+
+    @staticmethod
+    def add_args(parser: argparse.ArgumentParser) -> None:
+        parser.add_argument(
+            "--old_name",
+            dest="old_name",
+            required=True,
+            help="Full dotted name of object to rename. Eg: `foo.bar.baz`",
+        )
+
+        parser.add_argument(
+            "--new_name",
+            dest="new_name",
+            required=True,
+            help="Full dotted name of replacement object. Eg: `foo.bar.baz`",
+        )
+
+    def __init__(self, context: CodemodContext, old_name: str, new_name: str) -> None:
+        super().__init__(context)
+        self.old_name = old_name
+        self.new_name = new_name
+
+        old_names = self.old_name.rpartition(".")
+        new_names = self.new_name.rpartition(".")
+
+        # Mapping for easy lookup later.
+        self.equivalents: Dict[str, str] = {
+            old_names[2]: new_names[2],
+            "".join(old_names[:2]): new_names[0],
+        }
+
+        self.scheduled_imports = False
+
+    def visit_Import(self, node: cst.Import) -> None:
+        for import_alias in node.names:
+            alias_name = get_full_name_for_node(import_alias.name)
+            if alias_name is not None:
+                self.record_asname(import_alias, alias_name)
+
+                # If the import statement is exactly equivalent to the old name, replace it.
+                if alias_name == self.old_name:
+                    self.cleanup_imports(node=node, new_module=self.new_name)
+
+    def leave_ImportFrom(
+        self, original_node: cst.ImportFrom, updated_node: cst.ImportFrom
+    ) -> cst.ImportFrom:
+        module = updated_node.module
+        if module is not None:
+            imported_module_name = get_full_name_for_node(module)
+            names = original_node.names
+
+            if imported_module_name is None or not isinstance(names, Sequence):
+                return updated_node
+            elif imported_module_name == self.old_name:
+                # Simply rename the imported module on the spot.
+                return updated_node.with_changes(
```

Comment:
I noticed that according to [the coverage report](https://codecov.io/gh/Instagram/LibCST/pull/308/diff) no test actually hits this line?

(Test coverage of this PR is otherwise quite excellent!)